### PR TITLE
Add `--force` flag for validation on `shopify theme dev/pull/push/share` if users are running the command in a theme directory

### PR DESCRIPTION
**Pull request by Poitrin**
### WHY are these changes introduced?

- Fixes https://github.com/Shopify/shopify-cli/issues/2669
- Addition to https://github.com/Shopify/shopify-cli/pull/2680

We want to…
- ensure that partners execute `shopify theme serve/dev` in the _correct_ directory
- enforce the usage of the theme [architecture](https://shopify.dev/themes/architecture#directory-structure-and-component-types) directory structure

### WHAT is this pull request doing?

Add `--force` flag for validation on `shopify theme dev` if users are running the command in a theme directory.

### How to test your changes?

See https://github.com/Shopify/shopify-cli/pull/2680, but test with `shopify theme dev`, instead of `shopify theme serve`.
Make sure that your CLI 2.x version has the correct changes.

### Post-release steps

- [x] <s>Update shopify.dev docs, as `--force` flag has been added.</s> – flag is hidden, so no need to update docs f.t.m.

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

**Comment by mgmanzella** [comment]
👋 depending on when this merges, keep [this other PR](https://github.com/Shopify/cli/pull/743/files) in mind to allowlist `force` in dev.ts

**Review by mgmanzella — COMMENTED** [review]
> I'd get @MeredithCastile's input in this one.

+1 let's do this before proceeding

**Review by karreiro — APPROVED** [review]
Thank you, @Poitrin! 🚀 

+1 for introducing it as a hidden flag for now, then, if some partner need it (in a CI/CD context) we can suggest the usage. 

Still, we indeed don't need to assume the contract with users until we properly discuss it at https://github.com/Shopify/cli/issues/760 :)

Great stuff!

**Review by mgmanzella — APPROVED** [review]
🔥 🚀

File: packages/theme/src/cli/commands/theme/dev.ts [hunk 1]
```diff
@@ -62,10 +62,16 @@ export default class Dev extends ThemeCommand {
...
+      hidden: true,
+      char: 'f',
+      description: 'Proceed without confirmation, if current directory does not seem to be theme directory.',
+      env: 'SHOPIFY_FLAG_FORCE',
+    }),
   }
 
   // Tokens are valid for 120m, better to be safe and refresh every 90min
-  ThemeRefreshTimeouInMinutes = 90
+  ThemeRefreshTimeoutInMinutes = 90
```
Comment (Poitrin) [inline packages/theme/src/cli/commands/theme/dev.ts:74]: Typo ;-)

File: packages/theme/src/cli/commands/theme/dev.ts [hunk 2]
```diff
@@ -78,14 +84,15 @@ export default class Dev extends ThemeCommand {
...
+    const refreshThemeSessionInterval = setInterval(async () => {
       output.debug('Refreshing theme session...')
       controller.abort()
       controller = new abort.Controller()
       await this.execute(store, command, controller)
-    }, this.ThemeRefreshTimeouInMinutes * 60 * 1000)
+    }, this.ThemeRefreshTimeoutInMinutes * 60 * 1000)
 
     await this.execute(store, command, controller)
+    clearInterval(refreshThemeSessionInterval)
```
Comment (Poitrin) [inline packages/theme/src/cli/commands/theme/dev.ts:95]: Solves https://github.com/Shopify/shopify-cli/pull/2680#issuecomment-1309337789

**Review by matteodepalo — APPROVED** [review]

**Pull request merged**